_includes/tutorials/aggregating-minmax/kstreams/markup/dev/make-topology.adoc

```diff
@@ -0,0 +1,18 @@
+Then create the following file at `src/main/java/io/confluent/developer/AggregatingMinMax.java`. 
+
++++++
+<pre class="snippet"><code class="java">{% include_raw tutorials/aggregating-minmax/kstreams/code/src/main/java/io/confluent/developer/AggregatingMinMax.java %}</code></pre>
++++++
+
+Let's take a close look at the `buildTopology()` function, which uses the Kafka Streams DSL.
+
+Using the `https://kafka.apache.org/{{ site.ak_javadoc_version }}/javadoc/org/apache/kafka/streams/StreamsBuilder.html[StreamsBuilder]` parameter, which is the helper object that lets us build our topology, we can apply the following sequence of stages:
+
+1. Call the `stream()` function which creates a `https://kafka.apache.org/{{ site.ak_javadoc_version }}/javadoc/org/apache/kafka/streams/kstream/KStream.html[KStream]<String, MovieTicketSales>` object based on the stream of records from the `inputTopic` Kafka topic.
+
+2. Our use case requires we calculate minimum and maximum movie revenue _by year_.  The `https://kafka.apache.org/{{ site.ak_javadoc_version }}/javadoc/org/apache/kafka/streams/kstream/KStream.html#groupBy-org.apache.kafka.streams.kstream.KeyValueMapper-[groupBy]` function creates a `https://kafka.apache.org/{{ site.ak_javadoc_version }}/javadoc/org/apache/kafka/streams/kstream/KGroupedStream.html[KGroupedStream]` object.  `KGroupedStream` represents a 'grouped record stream' which allows us to apply aggregations over the records, grouped by a the key.  Here, we are specifying the movie's year of release as the record value on which to group. 
```

Comment:
We might want to mention something like 
```
Since you are changing the key, Kafka Streams will re-partition the data.   But ths is handled automatically for you, so it's a seamless operation.
```